helm/templates/prometheus/prometheus.configmap.yaml

```diff
@@ -13,12 +13,14 @@ data:
     groups:
       - name: Inactivity
         rules:
-          - alert: Inactivity session
+          - record: idletime_minutes
+            expr: (time() - max_over_time(timestamp(sum by (app) (increase(http_request_duration_seconds_count[2m])) > 4.5)[1d:5m])) / 60
```

Comment:
Instead of having a complex expression, which is Jupyter specific, we should make use of the official Jupyter API instead: https://jupyter-server.readthedocs.io/en/latest/developers/rest-api.html

As discussed, it should be checked if it can be integrated into the `metrics` endpoint of the Jupyter container. Alternatively, we can have our own metrics endpoint.

Reply:
I checked. The `last_activity` metric is about the activity of the Jupyter kernel. Also the status API requires authentication.

Instead I updated the metric to take into account the start time of the JupyterLab instance. Not it nicely counts idle time from the start of the session.